client/ayon_core/cli.py

```diff
@@ -167,16 +167,29 @@ def run(script):
 
     if not script:
         print("Error: missing path to script file.")
+        return
+
+    # Remove first argument if it is the same as AYON executable
+    # - Forwards compatibility with future AYON versions.
+    # - Current AYON launcher keep the arguments with first argument but
+    #     future versions might remove it.
+    first_arg = sys.argv[0]
+    if is_running_from_build():
+        comp_path = os.path.join(os.environ["AYON_ROOT"], "start.py")
     else:
-
-        args = sys.argv
-        args.remove("run")
-        args.remove(script)
-        sys.argv = args
-
-        args_string = " ".join(args[1:])
-        print(f"... running: {script} {args_string}")
-        runpy.run_path(script, run_name="__main__", )
+        comp_path = os.getenv("AYON_EXECUTABLE")
+    # Normalize paths for comparison
+    first_arg = os.path.normpath(first_arg).lower()
+    comp_path = os.path.normpath(comp_path).lower()
```

Comment:
Should this be `os.path.realpath` because the argument path _may_ be relative? Or is that not the case? Or use some sort of `pathlib.Path` comparison?

Reply:
Rather `os.path.abspath` but `pathlib` might be better option.

Reply:
Use `Path(...).resolve()` for comaprison. Didn't find better approach.

Reply:
Note that the resolve only works for existing paths it seems - it doesn't turn relative paths into absolute paths if the path does not exist - which might be unwanted side effect?

Reply:
I think that's ok in this case.